### PR TITLE
[IDF-43] feat: HTTP API integration and other improvements

### PR DESCRIPTION
**Pull request by khoroshilov-flo**
CHANGELOG:
- Add HTTP API usage examples
- Add HTTP REST API usage examples
- Add custom authorizer examples
- Update `serverless` and other related dependencies
- Change CRLF to LF in `.editorconfig`
- Update media-info lambda to use HTTP API event
- Update errors/ section
- Update helper/ section
- Update CORS related recourses for REST API
- Make small changes to the serverless config to work with new version of `serverless`
- Fix `deploy:local` cmd to work with new versions offline plugin
- Update README

### CODE REVIEW

**Comment by khoroshilov-flo** [comment]
I will check errors in circle CI

File: api/rest-api/handler.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// Remove example
+
+import { log } from '@helper/logger';
+import { Handler } from 'aws-lambda';
+import { errorHandler } from '../../helper/rest-api/error-handler';
```
Comment (azhivaeva-flo) [inline api/rest-api/handler.ts]: Replace with @helper/rest-api/error-handler

File: api/rest-api/handler.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import { APIGatewayLambdaEvent } from '../../interfaces/api-gateway-lambda.interface';
```
Comment (azhivaeva-flo) [inline api/rest-api/handler.ts]: Replace with @interfaces/api-gateway-lambda.interface

File: package.json [hunk 3]
```diff
@@ -91,6 +91,7 @@
   },
   "dependencies": {
     "@redtea/format-axios-error": "^2.0.2",
+    "@workflowwin/iconik-api": "2.2.1",
```
Comment (azhivaeva-flo) [inline package.json]: We don't need iconik-api in this project.
win.sls has it and should be used for WIN projects as a template

**Review by azhivaeva-flo — CHANGES_REQUESTED** [review]

**Comment by azhivaeva-flo** [comment]
> I will check errors in circle CI

If you remove the iconik-api package, it will solve problems with CI 🙂

**Review by azhivaeva-flo — APPROVED** [review]
Good job! 🙂

**Review by zaikin-andrew — APPROVED** [review]

**Comment by azhivaeva-flo** [comment]
@khoroshilov-flo I noticed that this version of Serverless has problems with environment variables like `${file(env.yml):${opt:stage, self:provider.stage}.CLIENT}` after deploy. Could you check it, please?

**Comment by khoroshilov-flo** [comment]
> @khoroshilov-flo I noticed that this version of Serverless has problems with environment variables like `${file(env.yml):${opt:stage, self:provider.stage}.CLIENT}` after deploy. Could you check it, please?

No problems after deploy
image

**Review by emahota-flo — APPROVED** [review]

**Pull request merged**